ui-tests/tests/widget.test.ts

```diff
@@ -179,6 +179,410 @@ ConnectorWidget()`)
 }
 
 
+const autoPopulatedFieldsEmbeddedDatabases = [
+  { label: "DuckDB", connectionName: "somealias", database: "somedb" },
+  { label: "SQLite", connectionName: "somealias", database: "somedb" },
+];
+
+for (const { label, connectionName, database, port, username, password, host} of autoPopulatedFieldsEmbeddedDatabases) {
+    test(`test only relevant fields are auto-populated in embedded dbs
+          if dropdown selection changes: ${label}`, async ({ page }) => {
+        await displayWidget(page);
+
+        await page.locator("#createNewConnection").click();
+        await page
+          .locator("#selectConnection")
+          .selectOption({ label: "PostgreSQL" });
+
+        await page.locator("#connectionName").fill("somealias");
+        await page.locator("#username").fill("someuser");
+        await page.locator("#password").fill("somepassword");
+        await page.locator("#host").fill("localhost");
+        await page.locator("#port").fill("3308");
+        await page.locator("#database").fill("somedb");
+        await page.locator("#selectConnection").selectOption({ label: label });
+
+        expect(
+          await page.locator(`#connectionName`).evaluate((select) => select.value)
+        ).toBe(connectionName);
+
+        expect(
+          await page.locator(`#database`).evaluate((select) => select.value)
+        ).toBe(database);
+
+        const username = page.locator("#username");
+        expect(await username.count()).toBe(0);
+
+        const password = page.locator("#password");
+        expect(await password.count()).toBe(0);
+
+        const host = page.locator("#host");
+        expect(await host.count()).toBe(0);
+
+        const port = page.locator("#port");
+        expect(await port.count()).toBe(0);
+      });
+}
+
+
+const autoPopulatedFields = [
+  { label: "MySQL", connectionName: "default", port: "3306", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "MariaDB", connectionName: "default", port: "3306", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Snowflake", connectionName: "default", port: "443", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Oracle", connectionName: "default", port: "1521", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "MSSQL", connectionName: "default", port: "1433", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
+  { label: "Redshift", connectionName: "default", port: "5439", username: "someuser",
+    password: "somepassword", host: "localhost", database: "somedb",
+  },
```

Comment:
across all tests, there's only a handful of values that change (e.g. all of these have the same database name, password, etc). so there's no need to parameterized over all values, just over the ones that change. this will make the code more readable

Reply:
[Fixed](https://github.com/ploomber/jupysql-plugin/pull/75/commits/05c49f46fff95133e1efae6e0dd1256252317214)